### PR TITLE
Single keypress switch enablement

### LINKED ISSUE #15: Push switches are difficult to select
in magit on emacs, I type `p` `-` `f` `u` and it does a force with lease push to origin. I think its okay if I have to do it differently in magit for vscode, but I think the typeahead search should perhaps be tweaked to:

1.) not search the descriptions
2.) be case sensitive (maybe?)

I've added a screenshot of what I see when I type `-f` to select a switch.

Screen Shot 2020-03-27 at 2 20 04 PM

This requires me to type more or select it with my arrow keys. Although, it looks like the arrow keys are always necessary, since I have to go down one to select my desired switch. Would be nice to eliminate that some how.

### PR DESCRIPTION
**Pull request by kahole**
Fixes #15

### CODE REVIEW

**Comment by stevenguh** [comment]
Just did a quick test, this doesn't allow a user to enter `-` in the switch menu.

**Comment by kahole** [comment]
It does now.

File: src/menu/menu.ts [hunk 1]
```diff
@@ -101,39 +102,63 @@ export class MenuUtil {
...
-        _quickPick.canSelectMany = true;
-        _quickPick.title = 'Switches (select with <space>)';
-      }
+      _quickPick.canSelectMany = true;
+      _quickPick.title = 'Switches (type shortname of switches you want to enable)';
+      _quickPick.items = menuState.switches.map(s => ({ label: s.shortName, detail: s.longName, description: `\t${s.description}`, picked: s.activated }));
+      _quickPick.selectedItems = _quickPick.items.filter(s => s.picked);
 
-      const acceptListenerDisposable = _quickPick.onDidAccept(async () => {
+      const eventListenerDisposable = _quickPick.onDidChangeValue(async (e) => {
```
Comment (stevenguh) [inline src/menu/menu.ts:116]: Some nit: I think `async` is not needed here and L:139

File: src/menu/menu.ts [hunk 2]
```diff
@@ -101,39 +102,63 @@ export class MenuUtil {
...
+        let quickPickValue = _quickPick.value;
+        let updatedItems: QuickPickItem[] = [];
 
-        menuState.switches!.forEach(s => {
-          updatedSwitches.push({ ...s, activated: _quickPick.selectedItems.find(item => item.label === s.shortName) !== undefined });
-        });
+        for (let item of _quickPick.items) {
+
+          let match = MenuUtil.matchesSwitch(quickPickValue, item.label);
+          updatedItems.push(({ ...item, picked: match ? !item.picked : item.picked }));
```
Comment (stevenguh) [inline src/menu/menu.ts:127]: updatedItems is not being used?

**Comment by stevenguh** [comment]
Love it. It looks much cleaner. Thank you for working on this.

File: src/menu/menu.ts (hunk 2 above)
Comment (kahole) [reply src/menu/menu.ts]: it was still work in progress you see. But now I think it should be good.

**Comment by kahole** [comment]
Thanks for helping :)

**Pull request merged**